specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json

```
@@ -0,0 +1,390 @@
+            "description": "Name of the Dapr Component.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DaprSecretsCollection"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List Container Apps Secrets": {
+            "$ref": "./examples/ConnectedEnvironmentsDaprComponents_ListSecrets.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConnectedEnvironmentDaprComponentsCollection": {
+      "description": "Collection of Dapr Components for Environments",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of Dapr component resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConnectedEnvironmentDaprComponent"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ConnectedEnvironmentDaprComponent": {
+      "description": "Dapr Component.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Dapr component properties",
+          "type": "object",
+          "allOf": [
+            {
+              "$ref": "./CommonDefinitions.json#/definitions/DaprComponentProperties"
+            }
+          ],
+          "properties": {
+            "provisioningState": {
+              "description": "Provisioning state of the Connected Environment Dapr Component.",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "Canceled",
+                "InProgress"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "ConnectedEnvironmentDaprComponentProvisioningState",
+                "modelAsString": true
+              }
+            },
+            "deploymentErrors": {
+              "description": "Any errors that occurred during deployment or deployment validation",
+              "type": "string",
+              "readOnly": true
+            }
+          },
+          "x-ms-client-flatten": true
```

And this can be simplified:
```suggestion
          "provisioningState": {
            "description": "Provisioning state of the Connected Environment Dapr Component.",
            "enum": [
              "Succeeded",
              "Failed",
              "Canceled",
              "InProgress"
            ],
            "type": "string",
            "readOnly": true,
            "x-ms-enum": {
              "name": "ConnectedEnvironmentDaprComponentProvisioningState",
              "modelAsString": true
            }
          },
          "deploymentErrors": {
            "description": "Any errors that occurred during deployment or deployment validation",
            "type": "string",
            "readOnly": true
          }
```